test/InSAR/InSARTest.jl

```diff
@@ -0,0 +1,120 @@
+    check_real = output_type <: Real
+
+    # check that the polynomial is not all zeros
+    check_non_zero = sum(output .!= 0) != 0;
+
+    # check if function is monotonically increasing/decreasing
+    check_monotonically_increasing = [output[i+1] >= output[i] for i in eachindex(output[1:end-1])];
+    check_monotonically_decreasing = [output[i+1] <= output[i] for i in eachindex(output[1:end-1])];
+    check_monotonic = all(check_monotonically_increasing) | all(check_monotonically_decreasing);
+
+    test_ok = all([check_real, check_non_zero, check_monotonic]);
+
+    if !test_ok
+        println([check_real, check_non_zero, check_monotonic])
+        println("Debug info: ", string(StackTraces.stacktrace()[1].func))
+        println("eltype(_doppler_fm_rate(x, param, x0)): ", output_type)
+        if !all(check_monotonically_increasing)
+            println("Index for breaking monotonic increase: ", findall(check_monotonically_increasing))
+        elseif !all(check_monotonically_decreasing)
+            println("Index for breaking monotonic decrease: ", findall(check_monotonically_decreasing))
+        else
+            println("Function is monotonic")
+        end
+    end
+
+    return test_ok
+end
+
+function test_phase_ramp(rows, columns, burst_number, mid_burst_speed, meta_data, orbit_states)
+    # check if output is real
+    ramp = SARProcessing.phase_ramp(rows, columns, burst_number, mid_burst_speed, meta_data, orbit_states);
+    output_type = eltype(ramp);
+    check_real = output_type <: Real
+
+    # check that the ramp is not all zeros
+    check_non_zero = sum(ramp .!= 0) != 0;
+
+    # check that length of each element is 1 for reramp * deramp
+    check_ramp_cancellation_length = all(length.(exp.(-ramp .* im) .* exp.(ramp .* im)) .== 1)
+
+    test_ok = all([check_real, check_non_zero, check_ramp_cancellation_length])
+    return test_ok
+end
+
+function test_deramp(image, rows, columns, burst_number, mid_burst_speed, meta_data, orbit_states)
+    ramp = SARProcessing.phase_ramp(rows, columns, burst_number, mid_burst_speed, meta_data, orbit_states);
+    deramped_image = SARProcessing.deramp(image, ramp)
+
+    output_type = eltype(deramped_image);
+    check_real = output_type <: Complex
+
+    # check that not all elements are zero
+    check_non_zero = sum(deramped_image .!= 0) != 0;
+
+    test_ok = all([check_real, check_non_zero])
+    return test_ok
+end
+
+function test_reramp(image, rows, columns, burst_number, mid_burst_speed, meta_data, orbit_states)
+    ramp = SARProcessing.phase_ramp(rows, columns, burst_number, mid_burst_speed, meta_data, orbit_states);
+    reramped_image = SARProcessing.reramp(image, ramp)
+    output_type = eltype(reramped_image);
+    check_real = output_type <: Complex
+
+    # check that not all elements are zero
+    check_non_zero = sum(reramped_image .!= 0) != 0;
+
+    test_ok = all([check_real, check_non_zero])
+    return test_ok
+end
+
+@testset "InSARTest.jl" begin
```

Comment:
This test set is set up very differently from everything else. 

Separate it out in test functions like other places and call them one at a time. You can also just hardcoded stuff related to the test data. We do not have to do a big loop over what is essential 1 burst

Reply:
I understand the loop part, I am not sure I understand what you want me to do with "Separate it out in test functions like other places and call them one at a time"? 
Should I make more tests?

Reply:
Make the @testset look some thing like this. 

<img width="512" alt="image" src="https://user-images.githubusercontent.com/30597266/210131524-5413ea84-685f-405c-b1bb-964cc459dbc0.png">

This is done by moving the logic into the test functions and not having it in the testset part. Remeber that 
- you can reuse code parts between test functions by placing it in a separate function and call it in the test functions
- You can check multiple conditions in one test by 
```
test_ok = condition1
test_ok &= condtion2
return test_ok
```